src/components/updatemenus/draw.js

```diff
@@ -362,11 +417,51 @@ function findDimenstions(gd, menuOpts) {
             tLines = tspans[0].length || 1,
             hEff = Math.max(tHeight * tLines, constants.minHeight) + constants.textOffsetY;
 
-        menuOpts.width = Math.max(menuOpts.width, wEff);
+        hEff = Math.ceil(hEff);
+        wEff = Math.ceil(wEff);
+
+        // Store per-item sizes since a row of horizontal buttons, for example,
+        // don't all need to be the same width:
+        menuOpts.widths[i] = wEff;
```

Comment:
Very nice. I think now this exact same `findDimensions` routine could be use for `RangeSelectors` - that will be for a future PR though.